# Bellabeat Case Study

# 1\. Defining A Problem

Bellabeat is a successful small company, but they have the potential to become a larger player in the global smart device market. Urška Sršen, cofounder and Chief Creative Officer of Bellabeat, believes that analyzing smart device fitness data could help unlock new growth opportunities for the company. You have been asked to focus on one of Bellabeat’s products and analyze smart device data to gain insight into how consumers are using their smart devices. The insights you discover will then help guide marketing strategy for the company. You will present your analysis to the Bellabeat executive team along with your high-level recommendations for Bellabeat’s marketing strategy.

### **Characters:**

- **Urška Sršen:** Bellabeat’s cofounder and Chief Creative Officer 
- **Sando Mur:** Mathematician and Bellabeat’s cofounder; key member of the Bellabeat executive team 
- **Bellabeat marketing analytics team:** A team of data analysts responsible for collecting, analyzing, and reporting data that helps guide Bellabeat’s marketing strategy. You joined this team six months ago and have been busy learning about Bellabeat’’s mission and business goals — as well as how you, as a junior data analyst, can help Bellabeat achieve them.

**A business task is defined by the next questions:**

1. What are some trends in smart device usage? 
2. How could these trends apply to Bellabeat customers? 
3. How could these trends help influence Bellabeat marketing strategy?

# 2\. Preparing Data

To solve the problem I am given a database 'bellabeat', which includes datasets:

1\. 'dailyActivity' (940 records, 15 fields), which describes users' daily activity statistics, such as daily distances walked, steps made, activity level times and distances, and calories burned:

In [8]:
SELECT COUNT(*) AS number_of_rows FROM dailyActivity;
SELECT TOP 5 * FROM dailyActivity;

(1 row affected)

(5 rows affected)

Total execution time: 00:00:00.068

number_of_rows 940

Id ActivityDate TotalSteps TotalDistance TrackerDistance LoggedActivitiesDistance VeryActiveDistance ModeratelyActiveDistance LightActiveDistance SedentaryActiveDistance VeryActiveMinutes FairlyActiveMinutes LightlyActiveMinutes SedentaryMinutes Calories 1503960366 4/12/2016 13162 8.5 8.5 0 1.87999999523163 0.550000011920929 6.05999994277954 0 25 13 328 728 1985 1503960366 4/13/2016 10735 6.96999979019165 6.96999979019165 0 1.57000005245209 0.689999997615814 4.71000003814697 0 21 19 217 776 1797 1503960366 4/14/2016 10460 6.73999977111816 6.73999977111816 0 2.44000005722046 0.400000005960464 3.91000008583069 0 30 11 181 1218 1776 1503960366 4/15/2016 9762 6.28000020980835 6.28000020980835 0 2.14000010490417 1.25999999046326 2.82999992370605 0 29 34 209 726 1745 1503960366 4/16/2016 12669 8.15999984741211 8.15999984741211 0 2.71000003814697 0.409999996423721 5.03999996185303 0 36 10 221 773 1863

2\. 'heartrateSeconds' (2483658 records, 3 fields) - shows all the users' heartrates recorded by the app, including user's ID, record date-time and the heartrate value:

In [7]:
SELECT COUNT(*) AS number_of_rows FROM heartrateSeconds;
SELECT TOP 5 * FROM heartrateSeconds;

(1 row affected)

(5 rows affected)

Total execution time: 00:00:00.774

number_of_rows 2483658

Id Time Value 2022484408 4/30/2016 1:16:15 PM 77 2022484408 4/30/2016 1:16:20 PM 79 2022484408 4/30/2016 1:16:25 PM 81 2022484408 4/30/2016 1:16:30 PM 80 2022484408 4/30/2016 1:16:40 PM 79

3\. 'sleepDay' (413 records, 5 fields) - about daily sleep time and time in bed of each user.

In [9]:
SELECT COUNT(*) AS numberOfRows FROM sleepDay;
SELECT TOP 5 * FROM sleepDay;

(1 row affected)

(5 rows affected)

Total execution time: 00:00:00.042

numberOfRows 413

Id SleepDay TotalSleepRecords TotalMinutesAsleep TotalTimeInBed 1503960366 4/12/2016 12:00:00 AM 1 327 346 1503960366 4/13/2016 12:00:00 AM 2 384 407 1503960366 4/15/2016 12:00:00 AM 1 412 442 1503960366 4/16/2016 12:00:00 AM 2 340 367 1503960366 4/17/2016 12:00:00 AM 1 700 712

4\. weightLogInfo (67 rows, 8 fields) - the data about users' weights and body mass indexes.

In [10]:
SELECT COUNT(*) AS numberOfRows FROM weightLogInfo;
SELECT TOP 5 * FROM weightLogInfo;

(1 row affected)

(5 rows affected)

Total execution time: 00:00:00.044

numberOfRows 67

Id Date WeightKg WeightPounds Fat BMI IsManualReport LogId 1503960366 5/2/2016 11:59:59 PM 52.5999984741211 115.963146545323 22 22.6499996185303 True 1462233599000 1503960366 5/3/2016 11:59:59 PM 52.5999984741211 115.963146545323 22.6499996185303 True 1462319999000 1927972279 4/13/2016 1:08:52 AM 133.5 294.317120016975 47.5400009155273 False 1460509732000 2873212765 4/21/2016 11:59:59 PM 56.7000007629395 125.002104340889 21.4500007629395 True 1461283199000 2873212765 5/12/2016 11:59:59 PM 57.2999992370605 126.324874550011 21.6900005340576 True 1463097599000

Then I used to check if all datasets are full in term of sample size (30 Bellabeat users should be included into testing and statistics):

In [11]:
SELECT COUNT(DISTINCT Id) AS dailyActivityIDs FROM dailyActivity;
SELECT COUNT(DISTINCT Id) AS heartrateIDs     FROM heartrateSeconds;
SELECT COUNT(DISTINCT Id) AS sleepDayIDs      FROM sleepDay;
SELECT COUNT(DISTINCT Id) AS weightLogIDs     FROM weightLogInfo;

(1 row affected)

(1 row affected)

(1 row affected)

(1 row affected)

Total execution time: 00:00:01.470

dailyActivityIDs 33

heartrateIDs 14

sleepDayIDs 24

weightLogIDs 8

Number of users in each dataset:

dailyActivity - 33

heartrateSeconds - 14

sleepDayIDs - 24

weightLogInfo - 8

Only 'dailyActivity' is full in terms of number of users (30+), so the analysis can be unfair.

  

# 3\. Cleaning

## 3.1. Cleaning 'dailyActivity'

### 3.1.1. Checking for duplicates

In [1]:
SELECT Id, ActivityDate, COUNT(*) AS numberOfRecords
FROM dailyActivity
GROUP BY Id, ActivityDate
HAVING COUNT(*) > 1

(0 rows affected)

Total execution time: 00:00:00.267

Id ActivityDate numberOfRecords

No duplicates.

3.1.2. Removing unuseful columns 

'SedentaryActivrDistance' column if wrong, because a human cannotwalk while being sedentary. LoggedActivitiesDistance is also unnesessary until almost all values in this column equal to 0.

In [2]:
ALTER TABLE dailyActivity 
DROP COLUMN SedentaryActiveDistance, LoggedActivitiesDistance;

Commands completed successfully.

Total execution time: 00:00:00.109

3.1.3. Rename column 'ActivityDate' to 'Date'

I used to do this for column name consistency between tables.

In [3]:
EXEC sp_rename 'dailyActivity.ActivityDate', 'Date', 'COLUMN';

Caution: Changing any part of an object name could break scripts and stored procedures.

Total execution time: 00:00:00.519

3.1.4. Checking datatypes

In [7]:
EXEC sp_help dailyActivity;

The object 'dailyActivity' does not have any indexes, or you do not have permissions.

No constraints are defined on object 'dailyActivity', or you do not have permissions.

No foreign keys reference table 'dailyActivity', or you do not have permissions on referencing tables.

No views with schema binding reference table 'dailyActivity'.

Total execution time: 00:00:00.633

Name Owner Type Created_datetime dailyActivity dbo user table 2022-07-23 19:39:25.867

Column_name Type Computed Length Prec Scale Nullable TrimTrailingBlanks FixedLenNullInSource Collation Id varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS Date varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS TotalSteps varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS TotalDistance varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS TrackerDistance varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS VeryActiveDistance varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS ModeratelyActiveDistance varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS LightActiveDistance varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS VeryActiveMinutes varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS FairlyActiveMinutes varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS LightlyActiveMinutes varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS SedentaryMinutes varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS Calories varchar no 50 yes no yes SQL_Latin1_General_CP1_CI_AS

Identity Seed Increment Not For Replication No identity column defined. NULL NULL NULL

RowGuidCol No rowguidcol column defined.

Data_located_on_filegroup PRIMARY

All the columns have varchar datatype, and that is incorrect. No columns should have varchar datatype:

\- 'Id', 'TotalSteps', 'VeryActiveMinutes', 'FairlyActiveMinutes', 'LightlyActiveMinutes', 'SedentaryMinutes' and 'Calories' - integer;

\- 'Date' - Date datatype;

\- 'TotalDistance', 'TrackerDistance', 'VeryActiveDistance', 'ModeratelyActiveDistance', 'LightActiveDistance' - double precision,

In [32]:
ALTER TABLE dailyActivity ALTER COLUMN Id bigint;
ALTER TABLE dailyActivity ALTER COLUMN Date DATE;
ALTER TABLE dailyActivity ALTER COLUMN TotalSteps INTEGER;
ALTER TABLE dailyActivity ALTER COLUMN TotalDistance FLOAT; 
ALTER TABLE dailyActivity ALTER COLUMN TrackerDistance FLOAT;
ALTER TABLE dailyActivity ALTER COLUMN VeryActiveDistance FLOAT; 
ALTER TABLE dailyActivity ALTER COLUMN ModeratelyActiveDistance FLOAT;
ALTER TABLE dailyActivity ALTER COLUMN LightActiveDistance FLOAT;
ALTER TABLE dailyActivity ALTER COLUMN VeryActiveMinutes INTEGER;
ALTER TABLE dailyActivity ALTER COLUMN FairlyActiveMinutes INTEGER;
ALTER TABLE dailyActivity ALTER COLUMN LightlyActiveMinutes INTEGER;
ALTER TABLE dailyActivity ALTER COLUMN SedentaryMinutes INTEGER;
ALTER TABLE dailyActivity ALTER COLUMN Calories INTEGER;

Commands completed successfully.

Total execution time: 00:00:00.512

3.1.5. Adding variable 'CaloriesLevel' 

This column will decribe the level of calories burned:

\- if Calories \< 2000, the CaloriesLevel = Low;

\- if Calories \>= 2000, then the CaloriesLevel = Normal.

In [34]:
ALTER TABLE dailyActivity ADD CaloriesLevel AS (CASE WHEN Calories < 2000 THEN 'Low' ELSE 'Normal' END);

Commands completed successfully.

Total execution time: 00:00:00.057

### 3.1.6. Checking for ouliers

In terms of mathematical statistic, outlier is a value that is not in range (average value +- 3\* standard deviation), so I used to check every variable for having outliers:

In [29]:
-- checking the number of records by date
WITH records AS 
(
    SELECT
        Date,
        COUNT(*) OVER (PARTITION BY Date ORDER BY Date) AS NumberOfRecords -- counting num of records made each day
    FROM dailyActivity
),
stats AS
(
    SELECT 
        STDEV(NumberOfRecords) AS StDeviation, -- counting standard deviation 
        AVG(NumberOfRecords) AS Average -- counting average value
    FROM records
)
-- defining a date outlier 
SELECT DISTINCT Date, NumberOfRecords
FROM records, stats
WHERE NumberOfRecords < Average - 3 * StDeviation OR NumberOfRecords > Average + 3 * StDeviation;
-- 2016-05-12 is an outlier with far less records made by users, because it was the last day of tracking

(1 row affected)

Total execution time: 00:00:00.083

Date NumberOfRecords 2016-05-12 21

In [32]:
-- checking the steps made each day
WITH stats AS
(
    SELECT 
        STDEV(TotalSteps) AS StDeviation, -- counting standard deviation 
        AVG(TotalSteps) AS Average -- counting average value
    FROM dailyActivity
)
-- defining a date outlier 
SELECT DISTINCT Id, Date, TotalSteps
FROM dailyActivity, stats
WHERE TotalSteps < Average - 3 * StDeviation OR TotalSteps > Average + 3 * StDeviation;
-- there are six outliers than need some additional analysis to define what to do with them

(6 rows affected)

Total execution time: 00:00:00.037

Id Date TotalSteps 1624580081 2016-05-01 36019 8053475328 2016-04-24 22988 8877689391 2016-04-12 23186 8877689391 2016-04-16 29326 8877689391 2016-04-27 23629 8877689391 2016-04-30 27745

In [34]:
-- checking the calories burned by users each date
WITH stats AS
(
    SELECT 
        STDEV(Calories) AS StDeviation, -- counting standard deviation 
        AVG(Calories) AS Average -- counting average value
    FROM dailyActivity
)
-- defining a date outlier 
SELECT DISTINCT Id, Date, Calories
FROM dailyActivity, stats
WHERE Calories < Average - 3 * StDeviation OR Calories > Average + 3 * StDeviation
ORDER BY Date;
-- there are 12 outliers 

(12 rows affected)

Total execution time: 00:00:00.062

Id Date Calories 8877689391 2016-04-16 4547 5577150313 2016-04-17 4552 6117666160 2016-04-21 4900 5577150313 2016-04-30 4501 8253242879 2016-04-30 0 5577150313 2016-05-01 4546 7007744171 2016-05-07 120 6290855005 2016-05-10 0 8792009665 2016-05-10 57 3977333714 2016-05-11 52 1503960366 2016-05-12 0 8583815059 2016-05-12 0

Well, according to https://www.sleepfoundation.org/how-sleep-works/how-your-body-uses-calories-while-you-sleep#:~:text=How%20Many%20Calories%20Do%20You,metabolic%20rate2%20(BMR).

Human burns at least 50 calories per hour even when doing nothing, sleeping and being sedentary. That's 1200 calories per day, so there's no way to burn less. I used to remove all the numbers less than 1200.

In [35]:
DELETE FROM dailyActivity WHERE Calories < 1200;

(18 rows affected)

Total execution time: 00:00:00.091

## 3.2. Cleaning 'heartrateSeconds' dataset

3.2.1. Changing datatypes

  

All the data in the table has 'varchar' data type, so we need to change it to the types corresponding to the data:

\- 'Id' - bigint;

\- 'Time' - datetime;

\- 'Value' - integer.

In [37]:
ALTER TABLE heartrateSeconds ALTER COLUMN Id bigint;
ALTER TABLE heartrateSeconds ALTER COLUMN Time datetime;
ALTER TABLE heartrateSeconds ALTER COLUMN Value INTEGER;

Commands completed successfully.

Total execution time: 00:04:52.515

3.2.2. Renaming column 'Value' to 'Heartrate'

In [38]:
EXEC sp_rename 'heartrateSeconds.Value', 'Heartrate', 'COLUMN';

Caution: Changing any part of an object name could break scripts and stored procedures.

Total execution time: 00:00:00.342

3.2.3. Adding variables 'Date', 'Hour' and 'Minute'

In [48]:
ALTER TABLE heartrateSeconds ADD 
    Date AS (CAST(Time AS date)),
    Hour AS DATEPART(HOUR, Time),
    Minute AS DATEPART(MINUTE, Time);

Commands completed successfully.

Total execution time: 00:00:00.003

3.2.4. Modifying table

There's no need to save multiple heartrate records per minute, so I used to round 'Time' to minutes

In [58]:
ALTER TABLE heartrateSeconds ADD Datetime AS FORMAT(Time, 'dd-MM-yyyy HH:mm') 

Commands completed successfully.

Total execution time: 00:00:00.004

Then, rather than maintain a table with multiple heartrates per minute, I used to create table with average heartrates per minute, it would contain 10x less records and memory.

In [64]:
SELECT
    Id,
    Datetime,
    Date,
    Hour,
    CAST(AVG(Heartrate) AS int) AS HeartRate
INTO heartrateMinutes
FROM heartrateSeconds
GROUP BY Id, Datetime, Date, Hour;

(333420 rows affected)

Total execution time: 00:01:36.531

### 3.2.5. Adding variable 'HeartrateLevel'

There are three levels of human's herartrate:

\- Low - less than 60 beats per minute (bpm);

\- Normal - 60-100 bpm;

\- High - more than 100 bpm.

  

I will include this information into the table:

In [65]:
ALTER TABLE heartrateMinutes ADD HeartrateLevel AS
(
    CASE 
        WHEN HeartRate < 60 THEN 'Low'     -- if a heartrate is lower than 60 bpm, then that's a low heartrate
        WHEN HeartRate > 100 THEN 'High'   -- if a heartrate is higher than 100 bpm, then that's a high heartrate
        ELSE 'Normal'                      -- if a heartrate is 60-100 bpm, then that's a normal heartrate
    END
);

Commands completed successfully.

Total execution time: 00:00:00.020

Then I used to delete 'heartrateSeconds' dataset to save memory:

In [66]:
DROP TABLE heartrateSeconds;

Commands completed successfully.

Total execution time: 00:00:00.031

### 3.2.6. Checking for duplicates

In [69]:
SELECT
    Id, 
    Datetime,
    COUNT(*) AS numOfDuplicates
FROM heartrateMinutes
GROUP BY Id, Datetime
HAVING COUNT(*) > 1; -- checking rows that have multiple entries = duplicates

(0 rows affected)

Total execution time: 00:00:01.228

Id Datetime numOfDuplicates

## 3.3. Processing 'sleepDay' dataset

### 3.3.1. Changing datatypes

In [71]:
ALTER TABLE sleepDay ALTER COLUMN Id bigint; -- changing datatype of Id from varchar to big integer
ALTER TABLE sleepDay ALTER COLUMN SleepDay Date; -- changing datatype from varchar to date
ALTER TABLE sleepDay ALTER COLUMN TotalSleepRecords int; -- from varchar to int
ALTER TABLE sleepDay ALTER COLUMN TotalMinutesAsleep int; -- from varchar to int
ALTER TABLE sleepDay ALTER COLUMN TotalTimeInBed int; -- from varchar to int

Commands completed successfully.

Total execution time: 00:00:00.111

### 3.3.2. Checking for duplicates

In [72]:
SELECT 
    Id,
    SleepDay,
    COUNT(*) AS NumOfDuplicates
FROM sleepDay
GROUP BY Id, SleepDay
HAVING COUNT(*) > 1; -- detect records that are repeated in the table (duplicates)

(3 rows affected)

Total execution time: 00:00:00.045

Id SleepDay NumOfDuplicates 8378563200 2016-04-25 2 4388161847 2016-05-05 2 4702921684 2016-05-07 2

There are 3 duplicate entries, so I must delete them.

In [73]:
WITH duplicateTable AS 
(
    SELECT *, 
        ROW_NUMBER() OVER -- count the number of unique entries
        (
            PARTITION BY Id, SleepDay -- group by primary key (Id + SleepDay) but not change the number of rows
            ORDER BY Id, SleepDay -- sort by primary key
        ) AS numOfRecords
    FROM sleepDay
)
DELETE FROM duplicateTable WHERE numOfRecords <> 1; -- delete all the non-unique entries with keeping the original

(3 rows affected)

Total execution time: 00:00:00.033

### 3.3.3. Rename 'SleepDay' column to 'Date'

In [76]:
EXEC sp_rename 'sleepDay.SleepDay', 'Date', 'COLUMN';

Caution: Changing any part of an object name could break scripts and stored procedures.

Total execution time: 00:00:00.549

### 3.3.4. Adding 'SleepRate' column

Daily sleeping norm for women is 7-9 hours (420-540 minutes) . that's normal level of sleeping. 

If less than 7 hours - 'Lack of dleep',

If more than 9 hours - 'Oversleeping'

In [77]:
ALTER TABLE sleepDay 
ADD SleepRate AS
(
    CASE 
    WHEN TotalMinutesAsleep < 420 THEN 'Lack of sleep'
    WHEN TotalMinutesAsleep > 540 THEN 'Oversleeping'
    ELSE 'Normal sleep'
    END
);

Commands completed successfully.

Total execution time: 00:00:00.069

### 3.3.5. Checking for outliers

In [38]:
-- checking the number of records by date
WITH stats AS
(
    SELECT 
        STDEV(TotalMinutesAsleep) AS StDeviation, -- counting standard deviation 
        AVG(TotalMinutesAsleep) AS Average -- counting average value
    FROM sleepDay
)
-- defining a date outlier 
SELECT DISTINCT Id, Date, TotalMinutesAsleep
FROM sleepDay, stats
WHERE TotalMinutesAsleep < Average - 3 * StDeviation OR TotalMinutesAsleep > Average + 3 * StDeviation;
-- 6 outliers than need an additional analysis

(6 rows affected)

Total execution time: 00:00:00.107

Id Date TotalMinutesAsleep 1644430081 2016-05-02 796 2320127002 2016-04-23 61 4319703577 2016-04-21 59 4388161847 2016-05-09 62 5553957443 2016-04-30 775 7007744171 2016-05-01 58

### 3.3.6. Adding 'TimeBeforeSleep' column

Which describes how much time we spend from going in bed and finally falling asleep.

In [39]:
ALTER TABLE sleepDay ADD TimeBeforeSleep AS (TotalTimeInBed - TotalMinutesAsleep);

Commands completed successfully.

Total execution time: 00:00:00.053

In [40]:
SELECT TOP 5 * FROM sleepDay;

(5 rows affected)

Total execution time: 00:00:00.027

Id Date TotalSleepRecords TotalMinutesAsleep TotalTimeInBed SleepRate TimeBeforeSleep 1503960366 2016-04-12 1 327 346 Lack of sleep 19 1503960366 2016-04-13 2 384 407 Lack of sleep 23 1503960366 2016-04-15 1 412 442 Lack of sleep 30 1503960366 2016-04-16 2 340 367 Lack of sleep 27 1503960366 2016-04-17 1 700 712 Oversleeping 12

## 3.4. Cleaning 'weightLogInfo'

### 3.4.1. Removing 'FAT' column

In [82]:
SELECT COUNT(*) AS NumberOfNAs FROM weightLogInfo 
WHERE FAT = ''; 

(1 row affected)

Total execution time: 00:00:00.018

NumberOfNAs 65

65/67 values of FAT column are empty, so the data in this column is incomplete, and I must remove the column.

In [3]:
ALTER TABLE weightLogInfo DROP COLUMN FAT;

Commands completed successfully.

Total execution time: 00:00:00.064

### 3.4.2. Changing data types of all columns

In [4]:
ALTER TABLE weightLogInfo ALTER COLUMN Id bigint; -- change 'Id' data type from varchar to big integer
ALTER TABLE weightLogInfo ALTER COLUMN Date date; -- change 'Date' data type from varchar to date
ALTER TABLE weightLogInfo ALTER COLUMN WeightKg float; -- change 'WeightKg' data type from varchar to float
ALTER TABLE weightLogInfo ALTER COLUMN WeightPounds float; -- change 'WeightPounds' data type from varchar to float
ALTER TABLE weightLogInfo ALTER COLUMN BMI float; -- change 'BMI' data type from varchar to float
ALTER TABLE weightLogInfo ALTER COLUMN IsManualReport bit; -- change 'IsManualReport' data type from varchar to bit
ALTER TABLE weightLogInfo ALTER COLUMN LogId bigint; -- change 'LogId' data type from varchar to big integer

Commands completed successfully.

Total execution time: 00:00:00.089

### 3.4.3. Checking for duplicates

In [6]:
SELECT 
    Id,
    Date,
    COUNT(*) AS NumOfDuplicates -- count the number of primary key (Id + Date) entries
FROM weightLogInfo
GROUP BY Id, Date
HAVING COUNT(*) > 1; -- output all the non-unique values

(0 rows affected)

Total execution time: 00:00:00.016

Id Date NumOfDuplicates

There's no duplicates.

### 3.4.4. Rename 'BMI' to 'BodyMassIndex'

BMI is an abbreviation, so not everybody can understand it. I used to name the column by its full name.

In [7]:
-- rename weightLogInfo column from BMI to BodyMassIndex using sp_rename function
EXEC sp_rename 'weightLogInfo.BMI', 'BodyMassIndex', 'COLUMN';

Caution: Changing any part of an object name could break scripts and stored procedures.

Total execution time: 00:00:00.538

### 3.4.4. Adding 'BodyFatRank' column

This column will rank womens' BMI according to the next criterias:

- 'Normal' BMI - 18.5-24.9 kg/m^2;
- 'Underweight' - less than 18.5 kg/m^2;
- 'Overweight' - more than 24.9 kg/m^2.

In [8]:
ALTER TABLE weightLogInfo ADD BodyFatRank AS -- add column 'BodyFatRank' which is calculated as:
(
    CASE
    WHEN BodyMassIndex < 18.5 THEN 'Underweight' -- if a BMI is lower than 18.4, then a person is underweight
    WHEN BodyMassIndex > 24.9 THEN 'Overweight' -- if a BMI is higher than 24.9, then a person is overweight
    ELSE 'Normal' -- if a BMI is 18.5-24.9, the person has normal body fat
    END
);

Commands completed successfully.

Total execution time: 00:00:00.013

# 4\. Analysis

## 4.1. Finding the start and end date of tracking in each dataset

In [10]:
SELECT MIN(Date) AS StartDate, MAX(Date) AS EndDate FROM dailyActivity;
SELECT MIN(Date) AS StartDate, MAX(Date) AS EndDate FROM sleepDay;
SELECT MIN(Date) AS StartDate, MAX(Date) AS EndDate FROM heartrateMinutes;
SELECT MIN(Date) AS StartDate, MAX(Date) AS EndDate FROM weightLogInfo;
-- in all tables tracking time is from 2016-05-12 to 2016-05-12

(1 row affected)

(1 row affected)

(1 row affected)

(1 row affected)

Total execution time: 00:00:00.234

StartDate EndDate 2016-04-12 2016-05-12

StartDate EndDate 2016-04-12 2016-05-12

StartDate EndDate 2016-04-12 2016-05-12

StartDate EndDate 2016-04-12 2016-05-12

## 4.2. 'dailyActivity' explorating

### Researching averages

In [41]:
-- average scores of each Id
SELECT
    Id,
    AVG(TotalSteps) AS AvgSteps,
    AVG(VeryActiveDistance) AS AvgVeryActiveDistance,
    AVG(VeryActiveMinutes) AS AvgVeryActiveMinutes,
    AVG(SedentaryMinutes) AS AvgSedentary,
    AVG(Calories) AS AvgCalories
FROM dailyActivity
GROUP BY Id;

(33 rows affected)

Total execution time: 00:00:00.093

Id AvgSteps AvgVeryActiveDistance AvgVeryActiveMinutes AvgSedentary AvgCalories 1644430081 7282 0,7300000009437403 9 1161 2811 7086361926 9557 2,8613333364327747 43 867 2611 4702921684 8572 0,41741935476180064 5 766 2965 3977333714 11337 1,6579310161800218 19 731 1564 6290855005 5851 0,08857142925262469 2 1184 2692 3372868164 6861 0,6294999971985812 9 1077 1933 8877689391 16040 6,637419362342162 66 1112 3420 2026352035 5743 0,0063333332538604665 0 679 1553 5577150313 8304 3,113666655619939 87 754 3359 4319703577 7510 0,28733333349227924 3 760 2097 4020332650 2323 0,14699999640385303 5 1254 2428 4057192912 3838 0,05249999836087225 0 1217 1973 8053475328 14763 8,514838741671657 85 1148 2945 4445114986 4796 0,5232258109796432 6 829 2186 6962181067 10001 1,670333339770635 23 680 2017 8792009665 1919 0,02571428607084919 1 1096 2030 2022484408 11370 2,4216128837677737 36 1112 2509 8378563200 8717 2,503548401978707 58 716 3436 1927972279 916 0,09580645085342482 1 1317 2172 2320127002 4785 0,1103333327919245 1 1233 1744 5553957443 8795 1,49366666773955 23 683 1913 7007744171 11776 2,5115999865531924 32 1093 2640 1503960366 12520 2,9536666830380756 40 828 1876 8583815059 7438 0,8246666766703131 10 1261 2823 4558609924 7685 0,5493548389884728 10 1093 2033 2873212765 7555 0,6761290309650283 14 1097 1916 4388161847 10813 1,7193548346479097 23 836 3093 6775888955 2541 0,7323999975621696 11 1329 2175 1624580081 5836 0,9706666707992535 8 1270 1499 1844505072 2666 0,0086666665971279 0 1223 1603 6117666160 7046 0,12821428477764124 1 796 2261 2347167796 10077 1,1217646938474737 14 727 2139 8253242879 6842 2,337222183744112 21 1278 1887

### Calories burned vs. Activity distances and total steps

In [43]:
-- checking the relation between active distances and calories burned
SELECT TOP 100
    Id,
    Date,
    Calories,
    VeryActiveDistance, 
    ModeratelyActiveDistance, 
    LightActiveDistance
FROM dailyActivity
ORDER BY Calories DESC;

(100 rows affected)

Total execution time: 00:00:00.082

Id Date Calories VeryActiveDistance ModeratelyActiveDistance LightActiveDistance 6117666160 2016-04-21 4900 0,980000019073486 0,400000005960464 5,61999988555908 5577150313 2016-04-17 4552 5,98000001907349 0,829999983310699 2,3199999332428 8877689391 2016-04-16 4547 13,2399997711182 1,21000003814697 10,710000038147 5577150313 2016-05-01 4546 5,30999994277954 1,44000005722046 3,24000000953674 5577150313 2016-04-30 4501 5,82999992370605 0,790000021457672 2,60999989509583 8877689391 2016-04-30 4398 21,6599998474121 0,0799999982118607 4,92999982833862 5577150313 2016-04-24 4392 7,65000009536743 2,15000009536743 1,98000001907349 5577150313 2016-04-16 4274 6,6399998664856 1,27999997138977 2,73000001907349 8378563200 2016-04-21 4236 6,90000009536743 0,819999992847443 4,28999996185303 8378563200 2016-04-14 4163 5,61999988555908 1,02999997138977 3,91000008583069 8378563200 2016-04-18 4157 5,05000019073486 0,560000002384186 5,19999980926514 8877689391 2016-05-07 4142 0,0799999982118607 0,959999978542328 6,98999977111816 8378563200 2016-04-19 4092 5,30000019073486 0,879999995231628 4,17999982833862 8378563200 2016-04-13 4079 4,96000003814697 0,649999976158142 4,21000003814697 8378563200 2016-04-22 4044 4,90999984741211 0,589999973773956 4,17999982833862 4388161847 2016-05-07 4022 9,44999980926514 2,76999998092651 5,32999992370605 5577150313 2016-04-14 4022 3,32999992370605 0,310000002384186 2,77999997138977 5577150313 2016-04-21 4018 3,65000009536743 1,6599999666214 2,77999997138977 8877689391 2016-05-06 4015 12,789999961853 0,28999999165535 6,15999984741211 5577150313 2016-04-15 4005 3,92000007629395 1,60000002384186 3,55999994277954 8378563200 2016-04-25 4005 5,05000019073486 0,870000004768372 3,92000007629395 4388161847 2016-05-08 3934 9,89000034332275 1,25999999046326 3,23000001907349 8877689391 2016-04-15 3934 0,0500000007450581 0,0500000007450581 7,01000022888184 8877689391 2016-04-12 3921 12,2200002670288 0,340000003576279 7,82000017166138 8877689391 2016-04-23 3891 0 0 7,40000009536743 4020332650 2016-05-04 3879 0,670000016689301 1,03999996185303 5,57999992370605 1644430081 2016-04-30 3846 0,629999995231628 3,14000010490417 9,46000003814697 5577150313 2016-05-10 3841 2,55999994277954 0,75 3,34999990463257 8877689391 2016-05-11 3832 13,2200002670288 0,409999996423721 5,8899998664856 8877689391 2016-04-27 3808 13,0699996948242 0,439999997615814 7,09999990463257 8877689391 2016-04-25 3802 2,36999988555908 0,0700000002980232 6,26999998092651 5577150313 2016-05-03 3795 4,51999998092651 0,150000005960464 3,5699999332428 8877689391 2016-04-14 3793 10,5500001907349 0,589999973773956 7,75 8378563200 2016-05-11 3788 1,38999998569489 0,589999973773956 5,26999998092651 8378563200 2016-04-20 3787 2,23000001907349 0,439999997615814 4,78000020980835 8378563200 2016-05-03 3784 1,3400000333786 1,05999994277954 4,5 8378563200 2016-05-05 3783 0,889999985694885 0,159999996423721 5,73999977111816 8378563200 2016-04-26 3763 7,51000022888184 0,920000016689301 4,42000007629395 6775888955 2016-04-20 3727 3,76999998092651 1,74000000953674 2,22000002861023 8378563200 2016-05-09 3721 1,26999998092651 0,660000026226044 4,71999979019165 8877689391 2016-05-09 3710 11,1000003814697 0,800000011920929 6,23999977111816 4702921684 2016-04-23 3691 0,759999990463257 3,24000000953674 8,27000045776367 4702921684 2016-05-07 3683 0,370000004768372 2,30999994277954 8,97000026702881 8877689391 2016-04-20 3679 11,0200004577637 0,689999997615814 6,34000015258789 8877689391 2016-04-19 3676 12,1499996185303 0,180000007152557 5,03000020980835 8378563200 2016-04-15 3666 4,90999984741211 1,14999997615814 5,40999984741211 8877689391 2016-04-21 3659 12,289999961853 0,419999986886978 4,8899998664856 4020332650 2016-04-12 3654 0,150000005960464 0,239999994635582 5,67999982833862 8378563200 2016-05-06 3644 1,54999995231628 0,25 3,77999997138977 8378563200 2016-04-12 3635 0,829999983310699 0,709999978542328 4,5 5577150313 2016-04-18 3625 4,8600001335144 0,720000028610229 1,8200000524520

### Calories burned vs. Activity minutes

In [44]:
-- checking a relation between activity minutes and calories burned
SELECT TOP 100 -- top 100 only for exploratory and memory saving
    Id,
    Date,
    Calories,
    VeryActiveMinutes, 
    FairlyActiveMinutes, 
    LightlyActiveMinutes,
    SedentaryMinutes
FROM dailyActivity
ORDER BY Calories DESC;

(100 rows affected)

Total execution time: 00:00:00.036

Id Date Calories VeryActiveMinutes FairlyActiveMinutes LightlyActiveMinutes SedentaryMinutes 6117666160 2016-04-21 4900 11 19 294 579 5577150313 2016-04-17 4552 200 37 159 525 8877689391 2016-04-16 4547 94 29 429 888 5577150313 2016-05-01 4546 194 72 178 499 5577150313 2016-04-30 4501 207 45 163 621 8877689391 2016-04-30 4398 124 4 223 1089 5577150313 2016-04-24 4392 210 65 141 425 5577150313 2016-04-16 4274 184 56 158 472 8378563200 2016-04-21 4236 137 16 145 677 8378563200 2016-04-14 4163 123 21 174 699 8378563200 2016-04-18 4157 117 10 174 720 8877689391 2016-05-07 4142 105 28 271 1036 8378563200 2016-04-19 4092 120 19 154 737 8378563200 2016-04-13 4079 116 14 169 680 8378563200 2016-04-22 4044 113 12 159 769 4388161847 2016-05-07 4022 120 56 260 508 5577150313 2016-04-14 4022 118 30 176 662 5577150313 2016-04-21 4018 110 74 175 670 8877689391 2016-05-06 4015 96 17 232 1095 5577150313 2016-04-15 4005 115 54 199 695 8378563200 2016-04-25 4005 117 16 141 692 4388161847 2016-05-08 3934 107 38 178 576 8877689391 2016-04-15 3934 106 1 281 1052 8877689391 2016-04-12 3921 85 7 312 1036 8877689391 2016-04-23 3891 102 6 300 1032 4020332650 2016-05-04 3879 13 46 346 531 1644430081 2016-04-30 3846 9 71 402 816 5577150313 2016-05-10 3841 104 37 194 639 8877689391 2016-05-11 3832 88 12 213 1127 8877689391 2016-04-27 3808 93 8 235 1104 8877689391 2016-04-25 3802 113 8 218 1101 5577150313 2016-05-03 3795 97 8 212 580 8877689391 2016-04-14 3793 68 13 298 1061 8378563200 2016-05-11 3788 72 10 184 763 8378563200 2016-04-20 3787 82 8 169 763 8378563200 2016-05-03 3784 71 20 195 822 8378563200 2016-05-05 3783 66 3 214 764 8378563200 2016-04-26 3763 90 18 161 593 6775888955 2016-04-20 3727 70 113 178 1079 8378563200 2016-05-09 3721 71 13 171 772 8877689391 2016-05-09 3710 73 19 217 1131 4702921684 2016-04-23 3691 9 66 408 469 4702921684 2016-05-07 3683 5 46 439 577 8877689391 2016-04-20 3679 73 19 225 1123 8877689391 2016-04-19 3676 82 13 214 1131 8378563200 2016-04-15 3666 60 23 190 729 8877689391 2016-04-21 3659 82 13 226 1119 4020332650 2016-04-12 3654 4 15 331 712 8378563200 2016-05-06 3644 74 5 166 831 8378563200 2016-04-12 3635 65 15 156 723 5577150313 2016-04-18 3625 114 32 130 623 8053475328 2016-05-08 3589 125 14 228 1073 8378563200 2016-05-10 3586 63 13 152 840 5577150313 2016-04-28 3580 99 51 141 692 8053475328 2016-04-24 3577 129 33 182 1096 8877689391 2016-04-13 3566 108 18 216 1098 8053475328 2016-04-23 3554 125 14 223 741 8877689391 2016-04-17 3545 58 15 307 1060 4388161847 2016-04-22 3544 77 25 220 945 5577150313 2016-04-29 3544 97 36 165 739 4702921684 2016-04-24 3538 15 95 281 542 8583815059 2016-04-29 3513 46 67 258 1069 5577150313 2016-04-19 3501 108 23 111 733 1644430081 2016-04-16 3493 51 42 212 1135 8378563200 2016-05-02 3491 63 4 142 802 8877689391 2016-05-04 3477 66 8 212 1154 8877689391 2016-04-24 3455 64 1 227 1148 4020332650 2016-05-05 3429 38 42 196 916 8877689391 2016-04-22 3427 61 2 236 1141 4388161847 2016-05-06 3425 33 45 262 1100 8053475328 2016-04-14 3411 119 5 193 1123 8053475328 2016-04-15 3410 132 8 158 1142 5577150313 2016-04-12 3405 86 16 140 728 1644430081 2016-05-07 3404 41 17 243 1139 8053475328 2016-04-25 3403 118 9 209 1104 4702921684 2016-04-17 3394 12 66 302 437 5577150313 2016-04-25 3374 61 38 214 743 8378563200 2016-05-01 3369 71 10 127 669 8378563200 2016-04-16 3363 64 21 142 563 8053475328 2016-05-07 3331 121 20 148 1076 5577150313 2016-04-22 3329 62 30 200 823 1644430081 2016-05-06 3328 11 6 369 1054 4702921684 2016-04-16 3328 3 24 335 556 6290855005 2016-05-01 3327 8 95 282 1055 1644430081 2016-05-03 3324 10 94 221 1115 4388161847 2016-04-23 3306 58 5 215 749 1644430081 2016-04-19 3300 5 58 278 1099 5577150313 2016-04-27 3294 63 16 190 773 4388161847 2016-05-03 3290 18 18 306 984 4388161847 2016-04-25 3288 11 31 301 1054 4702921684 2016-05-08 3287 9 125 192 1019 4388161847 2016-04-30 3283 14 43 300 537 4388161847 2016-04-18 3274 27 8 239 1017 8583815059 2016-05-05 3266 0 0 70 1370 1644430081 

### Calories level stats

Let's see the proportion between normal calories burned and low calories burned in the tracking period.

In [57]:
SELECT 
    CaloriesLevel,
    COUNT(*) AS Frequency
FROM dailyActivity
GROUP BY CaloriesLevel;

(2 rows affected)

Total execution time: 00:00:00.063

CaloriesLevel Frequency Normal 571 Low 351

There are 571 records with normal calories level, and 351 - with low calories level.

## BMI (Body mass index) level stats

In [4]:
SELECT DISTINCT Id, BodyFatRank, COUNT (*) AS Frequency -- count a number of records of each body fat level
FROM weightLogInfo
GROUP BY Id, BodyFatRank;

(8 rows affected)

Total execution time: 00:00:00.044

Id BodyFatRank Frequency 1503960366 Normal 2 2873212765 Normal 2 6962181067 Normal 30 1927972279 Overweight 1 4319703577 Overweight 2 4558609924 Overweight 5 5577150313 Overweight 1 8877689391 Overweight 24

5\\8 users are overweight, but that data can be not clear, because of 33 users tested the app, not 8 only.

## SleepDay stats

### Each user's sleep averages:

In [7]:
-- average sleep time, average time in bed, average time before sleep by each user
SELECT 
    Id,
    AVG(TotalMinutesAsleep) AS AverageSleepTime,
    AVG(TotalTimeInBed) AS AverageTimeInBed,
    AVG(TimeBeforeSleep) AS AverageTimeBeforeSleep
FROM sleepDay
GROUP BY Id
ORDER BY AverageSleepTime;

(24 rows affected)

Total execution time: 00:00:00.119

Id AverageSleepTime AverageTimeInBed AverageTimeBeforeSleep 2320127002 61 69 8 7007744171 68 71 3 4558609924 127 140 12 3977333714 293 461 167 1644430081 294 346 52 8053475328 297 301 4 4020332650 349 379 30 6775888955 349 369 19 1503960366 360 383 22 4445114986 385 416 31 4388161847 400 423 23 4702921684 417 438 20 1927972279 417 437 20 5577150313 432 460 28 8792009665 435 453 18 8378563200 445 485 40 2347167796 446 491 44 6962181067 448 466 18 7086361926 453 466 13 5553957443 463 505 42 4319703577 476 501 25 6117666160 478 510 31 2026352035 506 537 31 1844505072 652 961 309

### Sleep quality stats:

In [21]:
-- total sleep quality stats
SELECT 
    SleepRate,
    COUNT (*) AS Frequency
FROM sleepDay
GROUP BY SleepRate;

(3 rows affected)

Total execution time: 00:00:00.351

SleepRate Frequency Lack of sleep 181 Normal sleep 190 Oversleeping 39

There are 190 records of normal sleep, 181 - of the lack of sleep, and 39 - of oversleeping.

In [22]:
-- sleep quality stats by each user
SELECT 
    Id,
    SleepRate,
    COUNT(*) AS Frequency
FROM sleepDay
GROUP BY Id, SleepRate
ORDER BY Id;

(57 rows affected)

Total execution time: 00:00:00.112

Id SleepRate Frequency 1503960366 Lack of sleep 22 1503960366 Normal sleep 1 1503960366 Oversleeping 2 1644430081 Lack of sleep 3 1644430081 Oversleeping 1 1844505072 Oversleeping 3 1927972279 Lack of sleep 3 1927972279 Normal sleep 1 1927972279 Oversleeping 1 2026352035 Lack of sleep 1 2026352035 Normal sleep 23 2026352035 Oversleeping 4 2320127002 Lack of sleep 1 2347167796 Lack of sleep 4 2347167796 Normal sleep 10 2347167796 Oversleeping 1 3977333714 Lack of sleep 27 3977333714 Normal sleep 1 4020332650 Lack of sleep 5 4020332650 Normal sleep 3 4319703577 Lack of sleep 3 4319703577 Normal sleep 21 4319703577 Oversleeping 2 4388161847 Lack of sleep 9 4388161847 Normal sleep 11 4388161847 Oversleeping 3 4445114986 Lack of sleep 17 4445114986 Normal sleep 11 4558609924 Lack of sleep 5 4702921684 Lack of sleep 13 4702921684 Normal sleep 13 4702921684 Oversleeping 1 5553957443 Lack of sleep 13 5553957443 Normal sleep 11 5553957443 Oversleeping 7 5577150313 Lack of sleep 11 5577150313 Normal sleep 12 5577150313 Oversleeping 3 6117666160 Lack of sleep 5 6117666160 Normal sleep 9 6117666160 Oversleeping 4 6775888955 Lack of sleep 2 6775888955 Normal sleep 1 6962181067 Lack of sleep 8 6962181067 Normal sleep 21 6962181067 Oversleeping 2 7007744171 Lack of sleep 2 7086361926 Lack of sleep 5 7086361926 Normal sleep 18 7086361926 Oversleeping 1 8053475328 Lack of sleep 2 8053475328 Normal sleep 1 8378563200 Lack of sleep 13 8378563200 Normal sleep 14 8378563200 Oversleeping 4 8792009665 Lack of sleep 7 8792009665 Normal sleep 8

## HeartrateSeconds stats

### Average heartrates by each user:

In [9]:
SELECT 
    Id, CAST(AVG(Heartrate) AS INTEGER) AS AverageHeartrate
FROM heartrateMinutes
GROUP BY Id
ORDER BY AverageHeartrate;

(14 rows affected)

Total execution time: 00:00:00.219

Id AverageHeartrate 4388161847 63 5553957443 64 5577150313 65 8792009665 70 2347167796 73 6962181067 73 8877689391 74 2022484408 78 4558609924 78 4020332650 81 6117666160 81 7007744171 89 6775888955 89 2026352035 92

### Average daily heartrate stats by each user:

In [16]:
-- daily average heartrate statistics of each user
SELECT 
    Id, Date, CAST(AVG(Heartrate) AS INTEGER) AS AverageHeartrate
FROM heartrateMinutes
GROUP BY Id, Date
ORDER BY Id, Date;

(334 rows affected)

Total execution time: 00:00:00.540

Id Date AverageHeartrate 2022484408 2016-04-12 73 2022484408 2016-04-13 78 2022484408 2016-04-14 70 2022484408 2016-04-15 78 2022484408 2016-04-16 74 2022484408 2016-04-17 82 2022484408 2016-04-18 80 2022484408 2016-04-19 80 2022484408 2016-04-20 81 2022484408 2016-04-21 85 2022484408 2016-04-22 77 2022484408 2016-04-23 70 2022484408 2016-04-24 80 2022484408 2016-04-25 79 2022484408 2016-04-26 81 2022484408 2016-04-27 79 2022484408 2016-04-28 69 2022484408 2016-04-29 81 2022484408 2016-04-30 75 2022484408 2016-05-01 71 2022484408 2016-05-02 76 2022484408 2016-05-03 80 2022484408 2016-05-04 80 2022484408 2016-05-05 79 2022484408 2016-05-06 80 2022484408 2016-05-07 73 2022484408 2016-05-08 76 2022484408 2016-05-09 78 2022484408 2016-05-10 84 2022484408 2016-05-11 81 2022484408 2016-05-12 79 2026352035 2016-04-17 68 2026352035 2016-04-25 98 2026352035 2016-05-02 82 2026352035 2016-05-09 97 2347167796 2016-04-12 84 2347167796 2016-04-13 71 2347167796 2016-04-14 71 2347167796 2016-04-15 72 2347167796 2016-04-16 86 2347167796 2016-04-17 74 2347167796 2016-04-18 69 2347167796 2016-04-19 79 2347167796 2016-04-20 76 2347167796 2016-04-21 71 2347167796 2016-04-22 70 2347167796 2016-04-23 80 2347167796 2016-04-24 75 2347167796 2016-04-25 71 2347167796 2016-04-26 70 2347167796 2016-04-27 72 2347167796 2016-04-28 72 2347167796 2016-04-29 62 4020332650 2016-04-12 83 4020332650 2016-04-14 89 4020332650 2016-04-15 91 4020332650 2016-04-16 99 4020332650 2016-04-17 96 4020332650 2016-05-02 89 4020332650 2016-05-03 77 4020332650 2016-05-04 82 4020332650 2016-05-05 84 4020332650 2016-05-06 75 4020332650 2016-05-07 88 4020332650 2016-05-08 77 4020332650 2016-05-09 85 4020332650 2016-05-10 71 4020332650 2016-05-11 90 4020332650 2016-05-12 92 4388161847 2016-04-13 64 4388161847 2016-04-14 66 4388161847 2016-04-15 65 4388161847 2016-04-16 74 4388161847 2016-04-17 65 4388161847 2016-04-18 73 4388161847 2016-04-19 63 4388161847 2016-04-20 63 4388161847 2016-04-21 61 4388161847 2016-04-22 71 4388161847 2016-04-23 64 4388161847 2016-04-24 60 4388161847 2016-04-25 65 4388161847 2016-04-26 60 4388161847 2016-04-27 60 4388161847 2016-04-28 60 4388161847 2016-04-29 64 4388161847 2016-04-30 62 4388161847 2016-05-01 58 4388161847 2016-05-02 63 4388161847 2016-05-03 64 4388161847 2016-05-04 60 4388161847 2016-05-05 59 4388161847 2016-05-06 67 4388161847 2016-05-07 63 4388161847 2016-05-08 67 4388161847 2016-05-09 62 4388161847 2016-05-10 59 4388161847 2016-05-11 57 4388161847 2016-05-12 58 4558609924 2016-04-12 75 4558609924 2016-04-13 83 4558609924 2016-04-14 80 4558609924 2016-04-15 83 4558609924 2016-04-16 75 4558609924 2016-04-17 82 4558609924 2016-04-18 79 4558609924 2016-04-19 82 4558609924 2016-04-20 75 4558609924 2016-04-21 83 4558609924 2016-04-22 80 4558609924 2016-04-23 84 4558609924 2016-04-24 82 4558609924 2016-04-25 80 4558609924 2016-04-26 86 4558609924 2016-04-27 81 4558609924 2016-04-28 82 4558609924 2016-04-29 74 4558609924 2016-04-30 76 4558609924 2016-05-01 69 4558609924 2016-05-02 77 4558609924 2016-05-03 74 4558609924 2016-05-04 75 4558609924 2016-05-05 80 4558609924 2016-05-06 70 4558609924 2016-05-07 74 4558609924 2016-05-08 100 4558609924 2016-05-09 77 4558609924 2016-05-10 71 4558609924 2016-05-11 73 4558609924 2016-05-12 78 5553957443 2016-04-12 62 5553957443 2016-04-13 60 5553957443 2016-04-14 66 5553957443 2016-04-15 64 5553957443 2016-04-16 64 5553957443 2016-04-17 59 5553957443 2016-04-18 64 5553957443 2016-04-19 68 5553957443 2016-04-20 59 5553957443 2016-04-21 65 5553957443 2016-04-22 65 5553957443 2016-04-23 66 5553957443 2016-04-24 63 5553957443 2016-04-25 66 5553957443 2016-04-26 67 5553957443 2016-04-27 63 5553957443 2016-04-28 64 5553957443 2016-04-29 66 5553957443 2016-04-30 65 5553957443 2016-05-01 65 5553957443 2016-05-02 66 5553957443 2016-05-03 65 5553957443 2016-05-04 60 5553957443 2016-05-05 63 5553957443 2016-05-06 62 5553957443 2016-05-07 61 5553957443 2016-05-08 64 5553957443 2016-05-09 67 5553957443 2016-05-10 69

### HeartrateLevel by each user:

In [14]:
-- daily heartrate level stats by each user
SELECT 
    Id, 
    Date,
    HeartrateLevel,
    COUNT(*) AS Frequency
FROM heartrateMinutes
GROUP BY Id, Date, HeartrateLevel
ORDER BY Id, Date;

(923 rows affected)

Total execution time: 00:00:00.548

Id Date HeartrateLevel Frequency 2022484408 2016-04-12 High 62 2022484408 2016-04-12 Normal 562 2022484408 2016-04-12 Low 94 2022484408 2016-04-13 High 82 2022484408 2016-04-13 Normal 619 2022484408 2016-04-13 Low 55 2022484408 2016-04-14 High 18 2022484408 2016-04-14 Normal 596 2022484408 2016-04-14 Low 156 2022484408 2016-04-15 Normal 630 2022484408 2016-04-15 Low 24 2022484408 2016-04-15 High 56 2022484408 2016-04-16 High 47 2022484408 2016-04-16 Low 61 2022484408 2016-04-16 Normal 321 2022484408 2016-04-17 High 75 2022484408 2016-04-17 Low 25 2022484408 2016-04-17 Normal 603 2022484408 2016-04-18 Normal 674 2022484408 2016-04-18 High 81 2022484408 2016-04-18 Low 26 2022484408 2016-04-19 High 61 2022484408 2016-04-19 Normal 554 2022484408 2016-04-19 Low 22 2022484408 2016-04-20 Low 16 2022484408 2016-04-20 High 117 2022484408 2016-04-20 Normal 723 2022484408 2016-04-21 Low 61 2022484408 2016-04-21 High 175 2022484408 2016-04-21 Normal 552 2022484408 2016-04-22 Low 59 2022484408 2016-04-22 Normal 688 2022484408 2016-04-22 High 82 2022484408 2016-04-23 Normal 522 2022484408 2016-04-23 High 4 2022484408 2016-04-23 Low 83 2022484408 2016-04-24 Low 12 2022484408 2016-04-24 High 64 2022484408 2016-04-24 Normal 551 2022484408 2016-04-25 High 65 2022484408 2016-04-25 Low 85 2022484408 2016-04-25 Normal 380 2022484408 2016-04-26 Normal 598 2022484408 2016-04-26 High 119 2022484408 2016-04-26 Low 73 2022484408 2016-04-27 High 57 2022484408 2016-04-27 Low 37 2022484408 2016-04-27 Normal 529 2022484408 2016-04-28 Normal 671 2022484408 2016-04-28 High 8 2022484408 2016-04-28 Low 85 2022484408 2016-04-29 High 94 2022484408 2016-04-29 Low 58 2022484408 2016-04-29 Normal 535 2022484408 2016-04-30 High 18 2022484408 2016-04-30 Normal 710 2022484408 2016-04-30 Low 36 2022484408 2016-05-01 High 60 2022484408 2016-05-01 Normal 467 2022484408 2016-05-01 Low 173 2022484408 2016-05-02 Low 66 2022484408 2016-05-02 Normal 614 2022484408 2016-05-02 High 67 2022484408 2016-05-03 Normal 647 2022484408 2016-05-03 High 102 2022484408 2016-05-03 Low 48 2022484408 2016-05-04 Normal 687 2022484408 2016-05-04 Low 7 2022484408 2016-05-04 High 90 2022484408 2016-05-05 Normal 645 2022484408 2016-05-05 High 67 2022484408 2016-05-05 Low 6 2022484408 2016-05-06 High 54 2022484408 2016-05-06 Normal 702 2022484408 2016-05-07 Normal 278 2022484408 2016-05-07 Low 59 2022484408 2016-05-07 High 23 2022484408 2016-05-08 Low 12 2022484408 2016-05-08 Normal 228 2022484408 2016-05-08 High 24 2022484408 2016-05-09 Normal 734 2022484408 2016-05-09 Low 16 2022484408 2016-05-09 High 84 2022484408 2016-05-10 Low 6 2022484408 2016-05-10 Normal 578 2022484408 2016-05-10 High 82 2022484408 2016-05-11 Normal 489 2022484408 2016-05-11 High 87 2022484408 2016-05-11 Low 18 2022484408 2016-05-12 Normal 418 2022484408 2016-05-12 High 56 2022484408 2016-05-12 Low 17 2026352035 2016-04-17 Normal 40 2026352035 2016-04-25 High 32 2026352035 2016-04-25 Normal 60 2026352035 2016-05-02 Normal 79 2026352035 2016-05-02 High 3 2026352035 2016-05-09 High 83 2026352035 2016-05-09 Normal 127 2347167796 2016-04-12 Low 4 2347167796 2016-04-12 Normal 831 2347167796 2016-04-12 High 164 2347167796 2016-04-13 Normal 1363 2347167796 2016-04-13 High 43 2347167796 2016-04-13 Low 18 2347167796 2016-04-14 High 16 2347167796 2016-04-14 Normal 1362 2347167796 2016-04-14 Low 29 2347167796 2016-04-15 High 58 2347167796 2016-04-15 Low 38 2347167796 2016-04-15 Normal 1267 2347167796 2016-04-16 High 135 2347167796 2016-04-16 Normal 639 2347167796 2016-04-17 Normal 1379 2347167796 2016-04-17 Low 9 2347167796 2016-04-17 High 9 2347167796 2016-04-18 Low 126 2347167796 2016-04-18 Normal 1276 2347167796 2016-04-18 High 35 2347167796 2016-04-19 Normal 924 2347167796 2016-04-19 High 226 2347167796 2016-04-19 Low 92 2347167796 2016-04-20 Low 37 2347167796 2016-04-20 Normal 914 2347167796 2016-04-20 High 63 2347167796 2016-04-21 High 42 2347167796 2016-04-21 Normal 1340 2347167796 2016-04-21 Low 46 2347167796 2016-04-22 High 

In [15]:
-- total  heartrate stats by each user
SELECT 
    Id, 
    HeartrateLevel,
    COUNT(*) AS Frequency
FROM heartrateMinutes
GROUP BY Id, HeartrateLevel
ORDER BY Id;

(41 rows affected)

Total execution time: 00:00:00.155

Id HeartrateLevel Frequency 2022484408 Normal 17505 2022484408 High 2081 2022484408 Low 1496 2026352035 Normal 306 2026352035 High 118 2347167796 Normal 20276 2347167796 High 1117 2347167796 Low 1112 4020332650 Normal 9238 4020332650 Low 936 4020332650 High 1039 4388161847 Normal 19981 4388161847 High 721 4388161847 Low 17407 4558609924 Normal 24443 4558609924 Low 999 4558609924 High 1101 5553957443 Normal 27486 5553957443 High 605 5553957443 Low 12009 5577150313 Normal 14737 5577150313 High 3121 5577150313 Low 16901 6117666160 Low 685 6117666160 Normal 21887 6117666160 High 2182 6775888955 Low 12 6775888955 Normal 3567 6775888955 High 1498 6962181067 Low 2511 6962181067 Normal 37124 6962181067 High 1855 7007744171 High 4044 7007744171 Normal 15972 7007744171 Low 6 8792009665 High 644 8792009665 Normal 15673 8792009665 Low 3671 8877689391 Normal 20674 8877689391 Low 4530 8877689391 High 2150

### Heartrate level stats during a day

In [20]:
-- this table shows how the heartrate level changes during a day 
SELECT 
    Hour,
    HeartrateLevel,
    COUNT(*) AS Frequency
FROM heartrateMinutes
GROUP BY Hour, HeartrateLevel
ORDER BY Hour;

(72 rows affected)

Total execution time: 00:00:00.150

Hour HeartrateLevel Frequency 0 Low 3724 0 Normal 6672 0 High 112 1 Normal 6325 1 High 130 1 Low 3730 2 Low 3862 2 High 86 2 Normal 5852 3 Normal 5551 3 High 15 3 Low 4240 4 High 20 4 Low 4491 4 Normal 5116 5 Normal 5527 5 Low 4204 5 High 181 6 High 838 6 Low 2411 6 Normal 7708 7 Low 2240 7 High 835 7 Normal 10814 8 Low 3267 8 High 1561 8 Normal 10873 9 High 1916 9 Normal 11622 9 Low 3076 10 High 1817 10 Low 2788 10 Normal 12443 11 Low 2094 11 Normal 13725 11 High 1395 12 Normal 13877 12 High 1596 12 Low 1562 13 Low 1873 13 Normal 13581 13 High 1304 14 High 1424 14 Low 2217 14 Normal 13102 15 High 1064 15 Low 2402 15 Normal 13138 16 High 1585 16 Low 1554 16 Normal 13288 17 High 1585 17 Normal 13540 17 Low 1121 18 Low 1082 18 Normal 13610 18 High 1630 19 High 1482 19 Normal 13341 19 Low 1148 20 High 903 20 Normal 12337 20 Low 1356 21 High 415 21 Normal 10765 21 Low 1864 22 High 222 22 Normal 8731 22 Low 2755 23 Normal 7331 23 Low 3214 23 High 160

## Sedentary and active time vs. Sleep Quality

I used to find a relation between how much time do people spend sedentary and how much time they sleep then.

In [28]:
-- a relation between sedentary minutes and sleep time, and between total steps made and sleep time
SELECT 
    SedentaryMinutes, 
    TotalSteps,
    sleepDay.TotalMinutesAsleep
FROM dailyActivity
INNER JOIN sleepDay
ON (dailyActivity.Id = sleepDay.Id AND
dailyActivity.Date = sleepDay.Date)
ORDER BY SedentaryMinutes DESC;

(404 rows affected)

Total execution time: 00:00:01.015

SedentaryMinutes TotalSteps TotalMinutesAsleep 1265 3702 59 1222 1982 77 1219 2268 98 1193 3176 119 1167 1675 166 1155 4631 79 1153 7550 74 1142 5600 58 1129 5079 61 1121 3428 115 1106 4676 106 1096 7833 171 1076 19769 74 1062 9148 103 1058 3761 296 1053 838 437 1053 4053 235 1020 10218 62 1019 6543 123 1017 11009 99 991 6724 137 986 356 398 976 1282 391 950 4496 322 945 12139 82 944 6506 253 943 655 350 941 980 475 937 2483 363 935 10096 368 916 3385 328 916 11728 226 916 1831 415 902 1868 442 901 6530 425 900 1758 423 897 2923 337 896 9603 471 895 2064 388 890 9105 293 878 10074 428 866 5183 401 864 8585 377 863 5162 423 862 1551 508 858 10201 469 857 14070 247 855 4369 385 853 10055 442 853 1219 486 852 13559 213 851 4468 459 848 2283 439 845 11658 152 844 13743 126 843 10988 420 841 2072 439 840 2961 370 840 6582 342 839 6326 353 839 7243 332 838 9819 325 837 8758 499 837 5563 370 836 1251 484 835 12022 338 834 12461 322 834 1619 503 833 11992 331 833 13755 245 831 7045 323 831 2564 458 829 11107 469 824 2276 515 823 9172 338 822 8712 405 821 6017 506 818 10544 360 817 4068 391 816 18213 124 816 15103 273 815 11181 341 814 15355 277 812 12848 380 810 11584 354 809 4514 374 806 1320 531 804 7114 261 802 6064 351 801 7641 291 801 12346 399 798 14727 277 797 11388 250 797 6222 322 797 14365 492 796 10199 366 795 5245 339 787 9123 472 787 3276 429 785 2424 498 784 9810 354 783 3915 405 782 13154 366 781 11177 262 781 7804 405 781 11682 322 780 2946 468 777 6877 400 777 8856 274 776 10735 384 776 5077 432 775 15506 304 775 5862 364 774 5546 442 773 12669 340 773 8330 447 773 6805 424 772 7860 384 772 8382 359 769 12200 441 767 6175 527 767 11596 441 765 6108 421 764 5439 408 764 8567 381 763 9143 368 763 9388 381 762 9753 428 762 4493 419 761 9632 400 761 10080 460 761 10288 456 759 5325 354 759 5153 493 758 9469 390 758 3403 393 757 9572 451 757 8161 412 757 6708 457 756 3809 436 756 15482 357 754 10035 295 754 12159 334 752 5664 465 750 11886 412 749 13236 478 749 2467 477 749 7174 343 746 12207 383 746 8232 435 745 10538 347 745 8614 414 745 11317 514 745 6815 298 744 5813 451 744 3147 528 744 10645 333 743 6393 421 743 13566 427 741 13217 357 741 22359 331 741 5232 439 740 5709 565 739 2421 516 739 7924 459 738 9023 442 738 12414 219 737 13070 387 737 10052 440 734 3703 458 734 678 750 734 3800 462 733 7303 361 733 7047 432 733 12574 449 732 7439 525 732 10946 433 732 14371 361 732 5980 436 731 8911 355 731 10081 422 730 10602 369 730 9167 370 729 9454 327 729 14461 513 728 10144 479 728 8135 419 728 13162 327 728 2547 524 727 11393 421 726 9762 412 726 10387 485 725 7502 417 724 9769 409 724 12058 235 724 4744 506 723 7626 338 722 5454 411 721 4660 619 720 10465 452 720 5029 511 720 13630 398 718 7213 425 717 8793 454 717 3974 441 717 5908 443 716 3945 462 716 3609 527 716 7018 511 714 7193 349 714 6157 402 713 16520 230 712 8539 501 712 6093 332 712 2915 520 712 14673 404 711 7198 337 711 7623 353 711 5417 506 709 4188 541 709 6722 452 709 12764 450 709 10039 430 709 4500 533 708 11835 451 707 16358 418 706 6831 388 706 4414 503 706 10091 446 705 10129 445 704 3490 555 704 10181 329 703 11550 237 702 7150 336 701 10613 416 699 13318 424 697 10320 455 696 10553 421 695 12087 392 695 8221 490 695 15108 486 695 10677 481 693 17022 357 692 14331 419 692 9841 361 692 9930 433 692 9524 450 692 4081 467 692 12405 388 691 10414 383 689 16556 377 689 11404 442 688 13585 259 687 13238 237 687 3821 523 685 12390 388 684 11193 439 683 10147 442 682 6943 404 682 3758 796 680 12312 318 680 10780 286 680 10433 441 680 12386 447 680 7875 441 678 8247 500 677 15148 396 676 7359 531 676 10352 467 676 11677 323 676 13072 312 673 9461 319 673 3365 658 671 10688 472 671 8954 523 670 10830 414 669 9648 474 669 12770 285 669 11419 475 667 11611 453 667 6910 502 665 4193 511 663 4993 531 663 13459 283 662 8596 477 660 14112 310 659 10742 476 659 10725 425 655 12912 466 654 12109 394 653 9482 433 653 3335 545 651 4249 447 649 9899 505 648 6711 46

## BMI vs. SedentaryMinutes & TotalSteps

In [32]:
-- a relation between sedentary time, total steps made and body mass index
SELECT 
    SedentaryMinutes,
    TotalSteps,
    weightLogInfo.BodyMassIndex
FROM dailyActivity
INNER JOIN weightLogInfo ON (dailyActivity.Id = weightLogInfo.Id
AND dailyActivity.Date = weightLogInfo.Date)
ORDER BY BodyMassIndex;

(66 rows affected)

Total execution time: 00:00:00.417

SedentaryMinutes TotalSteps BodyMassIndex 1057 8859 21,4500007629395 720 7566 21,6900005340576 798 14727 22,6499996185303 816 15103 22,6499996185303 731 10081 23,8199996948242 654 12109 23,8199996948242 683 10147 23,8500003814697 689 11404 23,8899993896484 697 10320 23,8899993896484 621 12627 23,8899993896484 745 6815 23,8899993896484 709 4188 23,8899993896484 591 10524 23,9300003051758 634 10145 23,9599990844727 645 10762 23,9599990844727 659 10742 23,9599990844727 708 11835 23,9599990844727 659 10725 23,9599990844727 484 20031 24 720 5029 24 655 12912 24 717 5908 24 837 5563 24 862 1551 24,1000003814697 722 5454 24,1000003814697 637 13239 24,1000003814697 639 13928 24,1000003814697 709 6722 24,1700000762939 741 13217 24,2099990844727 548 5652 24,2399997711182 626 15448 24,2399997711182 607 12342 24,3500003814697 796 10199 24,3899993896484 770 8064 25,1399993896484 1154 18193 25,2600002288818 1119 19377 25,2900009155273 1061 21129 25,3099994659424 1098 15337 25,4099998474121 1123 19948 25,4099998474121 1207 9733 25,4099998474121 1189 10818 25,4099998474121 1095 21727 25,4400005340576 1157 11101 25,4899997711182 1143 14890 25,4899997711182 1131 18785 25,5300006866455 1226 10930 25,5300006866455 1174 10686 25,5599994659424 1127 21420 25,5599994659424 1101 12986 25,5599994659424 1104 23629 25,5599994659424 1032 11200 25,5900001525879 1148 16674 25,5900001525879 888 29326 25,5900001525879 1089 27745 25,5900001525879 1131 20226 25,6100006103516 1215 11423 25,6800003051758 1036 23186 25,6800003051758 1065 11451 27 1057 7891 27,0400009155273 1080 8940 27,25 1121 3428 27,3199996948242 1106 10429 27,3799991607666 1363 29 27,4500007629395 993 8095 27,4599990844727 525 12231 28 986 356 47,5400009155273

## Sedentary Minutes vs. Heartrate levels

In [35]:
-- a relation between how much time people spend sedentary and how it affects their heart
WITH dailyHeartrate AS
(
    SELECT 
        Id, 
        Date,
        HeartrateLevel,
        COUNT(*) AS Frequency
FROM heartrateMinutes
GROUP BY Id, Date, HeartrateLevel
)
SELECT 
    SedentaryMinutes,
    dailyHeartrate.HeartrateLevel,
    dailyHeartrate.Frequency
FROM dailyActivity
INNER JOIN dailyHeartrate ON
(dailyActivity.Id = dailyHeartrate.Id AND
 dailyActivity.Date = dailyHeartrate.Date)
ORDER BY SedentaryMinutes DESC;

(912 rows affected)

Total execution time: 00:00:00.450

SedentaryMinutes HeartrateLevel Frequency 1440 Normal 8 1440 Normal 7 1440 High 3 1440 High 1 1438 Normal 2 1438 High 2 1437 High 14 1437 Normal 60 1431 High 3 1431 Normal 28 1420 High 11 1420 Normal 39 1410 Low 8 1410 High 16 1410 Normal 24 1410 High 30 1410 Normal 84 1400 High 41 1400 Normal 6 1392 High 37 1392 Normal 19 1377 High 63 1377 Normal 8 1360 Low 64 1360 Normal 776 1360 High 5 1350 High 44 1350 Normal 171 1347 High 112 1347 Normal 59 1344 High 69 1344 Normal 217 1335 Low 132 1335 Normal 269 1321 High 127 1321 Normal 137 1321 Low 2 1318 High 129 1318 Normal 230 1305 Normal 228 1305 Low 12 1305 High 24 1304 Normal 275 1304 Low 255 1304 High 50 1302 Low 1 1302 High 70 1302 Normal 180 1299 Normal 343 1299 Low 15 1299 High 175 1296 Normal 301 1296 High 31 1287 Normal 347 1287 High 28 1279 Normal 233 1279 High 109 1275 Normal 720 1275 Low 484 1275 High 26 1269 High 23 1269 Low 59 1269 Normal 278 1256 Normal 634 1256 High 21 1240 High 55 1240 Normal 807 1238 Low 150 1238 High 1 1238 Normal 652 1238 High 43 1238 Normal 620 1237 High 47 1237 Low 61 1237 Normal 321 1233 High 124 1233 Normal 218 1226 High 29 1226 Normal 645 1226 Low 98 1225 Normal 524 1225 Low 15 1225 High 44 1222 Normal 233 1222 High 155 1218 Normal 738 1218 Low 2 1218 High 2 1215 Low 366 1215 Normal 494 1215 High 16 1207 Low 361 1207 High 17 1207 Normal 603 1204 Normal 282 1204 High 71 1191 High 4 1191 Low 83 1191 Normal 522 1191 Low 104 1191 Normal 738 1191 High 15 1190 Normal 328 1190 High 106 1189 Normal 669 1189 High 24 1189 Low 258 1188 High 5 1188 Low 2 1188 Normal 802 1187 Low 77 1187 High 29 1187 Normal 767 1182 Normal 622 1182 High 10 1175 Normal 737 1175 Low 384 1175 High 11 1174 High 16 1174 Normal 798 1174 Low 32 1172 Low 143 1172 Normal 741 1172 High 39 1171 Normal 380 1171 High 65 1171 Low 85 1170 Low 56 1170 High 39 1170 Normal 781 1161 High 64 1161 High 3 1161 Low 12 1161 Normal 551 1161 Normal 767 1161 Low 20 1157 High 62 1157 Low 254 1157 Low 94 1157 Normal 562 1157 High 14 1157 Normal 696 1155 Normal 858 1155 High 35 1154 High 64 1154 Low 12 1154 Normal 794 1153 Normal 438 1153 Low 269 1153 High 65 1152 High 61 1152 Normal 554 1152 Low 22 1148 High 18 1148 Low 30 1148 High 69 1148 Normal 596 1148 Low 156 1148 Normal 752 1143 High 39 1143 Normal 453 1143 Low 440 1142 Low 2 1142 High 60 1142 Normal 467 1142 Low 173 1142 Normal 644 1142 High 95 1141 High 70 1141 Low 204 1141 Normal 684 1136 Normal 757 1136 High 119 1135 High 87 1135 Normal 738 1133 Normal 489 1133 High 87 1133 Low 18 1131 Low 234 1131 Normal 830 1131 Low 73 1131 High 74 1131 Normal 609 1131 High 83 1131 Normal 762 1131 Low 1 1131 High 98 1127 Normal 666 1127 Low 164 1127 High 99 1123 Low 85 1123 High 79 1123 Normal 671 1123 Normal 679 1123 Low 168 1123 High 8 1122 High 67 1122 Normal 630 1122 Low 2 1122 High 49 1122 Low 24 1122 Normal 835 1122 Normal 848 1122 Low 10 1122 Normal 871 1122 High 56 1121 High 1 1121 Low 108 1121 Normal 680 1119 High 57 1119 Normal 529 1119 High 87 1119 Normal 614 1119 Low 69 1119 Normal 783 1119 Low 37 1119 High 67 1119 Low 66 1116 High 23 1116 Normal 724 1116 Low 36 1112 Normal 645 1112 Low 11 1112 Low 6 1112 High 140 1112 High 67 1112 Normal 754 1112 High 9 1112 Normal 984 1111 High 20 1111 Normal 1044 1111 Low 55 1110 High 13 1110 Normal 845 1105 Low 59 1105 Low 7 1105 High 90 1105 Normal 687 1105 High 82 1105 Normal 688 1104 Normal 592 1104 Low 247 1104 High 54 1104 High 97 1104 Low 48 1104 Normal 647 1104 High 102 1104 Normal 702 1101 High 124 1101 Normal 726 1101 Normal 605 1101 Low 223 1101 High 172 1100 Normal 619 1100 High 82 1100 Normal 813 1100 High 28 1100 Low 55 1100 Low 382 1099 High 39 1099 Low 239 1099 Normal 445 1098 Low 222 1098 Normal 535 1098 High 112 1098 High 94 1098 Normal 592 1098 Low 58 1096 Low 1 1096 High 209 1096 High 29 1096 Normal 709 1096 Normal 813 1096 Low 64 1095 High 137 1095 Low 104 1095 Normal 672 1089 Low 80 1089 High 128 1089 Normal 651 1087 High 13 1087 Normal 794 1086 Low 73 1086 High 119 1086 Normal 598 1